pkg/util/metric/metric.go

```diff
@@ -383,12 +387,9 @@ type Histogram struct {
 	// it up right now. It should be doable though, since there is only one
 	// consumer of windowed histograms - our internal timeseries system.
 	windowed struct {
-		// prometheus.Histogram is thread safe, so we only
-		// need an RLock to record into it. But write lock
-		// is held while rotating.
-		syncutil.RWMutex
 		*tick.Ticker
-		prev, cur prometheus.HistogramInternal
+		syncutil.Mutex
+		prev, cur atomic.Value
```

Comment:
Not sure if this is possible, but can you use `atomic.Pointer[prometheus.HistogramInterval]` to avoid the type assertions?

https://github.com/paleozogt/go/commit/ffe48e00adf3078944015186819a1ed5c6aa8bec#diff-74356a5476466d442c2023f47d58e830715ba62eacd24b331200899a9cc3909cR39

Reply:
`prometheus.HistogramInterval` is an interface so I don't think I can use it as a pointer in this scenario